### ISSUE TITLE
Pipeline docker

### ISSUE THREAD

**Issue by CodingAdam** [UNLV-CS472-672/2024-S-GROUP8-CW#7]
The Unity CI Pipeline is erroring out a "Unable to find image 'unityci/editor:ubuntu-2022.3.20f1-windows-mono-1' locally",
being unable to dock the unity instance. The only post I could fine online as of the moment is switching the unity-builder to v4. 

I'm not sure if this will work, so as of now I will put this up as a issue for other to look into.

**Comment by GNAR1ZARD** [comment]
Ill def take a look after the portfolio pt1 is done

**Comment by GNAR1ZARD** [comment]
The issue at hand is a failed build due to a licensing problem. Despite a `UNITY_LICENSE` secret being set up in the repo, the automated build process is unable to detect or utilize the provided Unity license, resulting in an error. This impedes the build-and-test job within the CI pipeline from completing successfully. Additionally, there are warnings about the use of deprecated Node.js versions in the workflow. The primary task is to ensure the Unity license is valid, correctly formatted, and properly referenced by the workflow.

Im about to make a PR to update any outdated GitHub Actions to their latest supported versions.

**Comment by GNAR1ZARD** [comment]
Update: weird ok. Now I see the Do
![image](https://github.com/UNLV-CS472-672/2024-S-GROUP8-CW/assets/95197857/d461fa34-d177-4f31-83c3-a1207c44c70c)
cker issue 

![image](https://github.com/UNLV-CS472-672/2024-S-GROUP8-CW/assets/95197857/11199044-6a63-4ebf-a422-a111e36e08a2)

**Comment by GNAR1ZARD** [comment]
Alrighty Adam, I've done everything I can think of... I had an issue after updating the workflow to contain the updated secrets but then there seems to be an issue with the Docker container. The error message indicating that /usr/bin/docker failed with exit code 133 is a generic one and doesn't provide specifics about the root cause of the failure so then I added in a Debug step for Docker to try to troubleshoot further; given that the debug information about Docker doesn’t show any immediate issues, and the system appears to have over 15GB of memory, which should generally be sufficient, I'm stumped. @CodingAdam 

You guys know I have no expertise in Unity and I've tried very hard to figure out this issue.
